sched/signal/sig_dispatch.c

```diff
@@ -362,7 +363,7 @@ int nxsig_tcbdispatch(FAR struct tcb_s *stcb, siginfo_t *info)
 
       flags = enter_critical_section();
       if (stcb->task_state == TSTATE_WAIT_SIG &&
-          sigismember(&stcb->sigwaitmask, info->si_signo))
+          nxsig_ismember(&stcb->sigwaitmask, info->si_signo))
```

Comment:
There are lots checks here like this that do not account for failure return values.  They will all be treated as TRUE, that the signal is a member of the set.

Reply:
In this case signo was already tested by the first call to nxsig_ismember.  If we are executing this code, signo shall be valid.